pipelines/controller-cli/src/main.py

```diff
@@ -0,0 +1,245 @@
+        process_response(response, args)
+    except requests.exceptions.RequestException as e:
+        print(f"Error processing: {e}")
+
+
+def download_file(url: str, filename: str) -> str:
+    try:
+        with requests.get(url, stream=True) as r:
+            with open(filename, "wb") as f:
+                shutil.copyfileobj(r.raw, f)
+        return f"File downloaded successfully to {filename}"
+    except requests.exceptions.RequestException as e:
+        return f"Error downloading file: {e}"
+
+
+def logs(args) -> str:
+    try:
+        if args.download:
+            filename = args.filename if args.filename else "error.log"
+            response = download_file(f"{args.url}/download-error-log", filename)
+            process_response(response, args)
+        else:
+            print(
+                "You can use the tail command to watch the {DWH_ROOT}/error.log file."
+            )
+    except requests.exceptions.RequestException as e:
+        print(f"Error processing: {e}")
+
+
+def delete_snapshot(args) -> str:
+    try:
+        response = requests.delete(f"{args.url}/dwh?snapshotId={args.snapshot_id}")
+        if response.status_code == 204:
+            return f"Snapshot deleted successfully"
+        else:
+            return f"Error deleting snapshot: Status code {response.status_code}"
+    except requests.exceptions.RequestException as e:
+        return f"Error deleting snapshot: {e}"
+
+
+def dwh(args) -> str:
+    try:
+        if hasattr(args, "snapshot_id"):
+            response = delete_snapshot(args)
+        else:
+            response = _make_api_request(HTTP_GET, f"{args.url}/dwh")
+        process_response(response, args)
+    except requests.exceptions.RequestException as e:
+        print(f"Error processing: {e}")
+
+
+def main():
+    logging.basicConfig(level=logging.INFO)
+    parser = argparse.ArgumentParser(
+        description="The CLI tool for fhir-data-pipes",
+        formatter_class=argparse.RawTextHelpFormatter,
+    )
+    parser.add_argument(
+        "url",
+        type=str,
+        help="url of the pipeline controller's REST API",
+    )
+    subparsers = parser.add_subparsers(
+        dest="command",
+        help=f'{", ".join(str(value) for value in COMMAND_MAP.values())} '
+        "are the available commands.",
+        required=True,
+    )
+
+    config_parser = subparsers.add_parser("config", help="show config values")
+    config_parser.add_argument(
+        "--config-name", "-cn", required=False, help="name of the configuration key"
+    )
+    config_parser.set_defaults(func=config)
+
+    next_parser = subparsers.add_parser("next", help="show the next scheduled run")
+    next_parser.set_defaults(func=next)
+
+    status_parser = subparsers.add_parser(
+        "status", help="show the status of the pipeline"
+    )
+    status_parser.set_defaults(func=status)
+
+    run_modes = ", ".join(mode.lower() for mode in RUN_MODES)
```

Comment:
It seems that we always use the lower case of `RUN_MODES`, right? If yes, why not store lower case in the first place and remove the `lower()` calls?